### PR TITLE
Data resilience card for ceph storage dashboard

### PR DESCRIPTION
**Pull request by cloudbehl**
Signed-off-by: Ankush Behl <cloudbehl@gmail.com>

Screenshot from 2019-06-28 15-26-29

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/plugin.ts [hunk 1]
```diff
@@ -22,6 +29,24 @@ const plugin: Plugin<ConsumedExtensions> = [
...
+    },
+  },
+  {
+    type: 'Dashboards/Card',
+    properties: {
+      tab: 'persistent-storage',
+      position: GridPosition.MAIN,
+      loader: () =>
+        import(
+          './components/dashboard-page/storage-dashboard/data-resilience/data-resilience' /* webpackChunkName: "data-resilence-card" */
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/plugin.ts]: Please add `ceph` prefix to webpack chunk name (there is also missing `i` in `data-resilence-card"`
```suggestion
          './components/dashboard-page/storage-dashboard/data-resilience/data-resilience' /* webpackChunkName: "webpack-data-resilience-card" */
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resilience/data-resilience.scss [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  justify-content: center;
+  text-align: center;
+}
+
+.ceph-data-resiliency__utilization-bar {
+  text-align: left;
+}
+
+.ceph-data-resiliency__icon-ok {
+  color: yellowgreen;
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resilience/data-resilience.scss]: please use PF color variable 
```suggestion
  color: $pf-color-light-green-400;
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resilience/data-resilience.scss [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+.ceph-data-resiliency__icon-error {
+  color: #c9190b;
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resilience/data-resilience.scss]: PF color variable

```suggestion
  color: $pf-color-red-100;
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resilience/data-resilience.scss [hunk 4]
```diff
@@ -0,0 +1,32 @@
+.ceph-data-resiliency__title {
```
Comment (spadgett) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resilience/data-resilience.scss]: alphbetize classes and styles throughout

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resilience/data-resilience.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    Object.keys(DATA_RESILIENCE_QUERIES).forEach((key) =>
+      watchPrometheus(DATA_RESILIENCE_QUERIES[key]),
+    );
+    return () =>
+      Object.keys(DATA_RESILIENCE_QUERIES).forEach((key) =>
+        stopWatchPrometheusQuery(DATA_RESILIENCE_QUERIES[key]),
+      );
+  }, [watchPrometheus, stopWatchPrometheusQuery]);
+
+  const cleanAndActivePgRaw = prometheusResults.getIn([
```
Comment (spadgett) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resilience/data-resilience.tsx]:
```suggestion
  const cleanAndActivePGRaw = prometheusResults.getIn([
```

here and below

File: frontend/packages/ceph-storage-plugin/src/plugin.ts [hunk 6]
```diff
@@ -1,10 +1,17 @@
 import * as _ from 'lodash';
 
-import { Plugin, ModelFeatureFlag, ModelDefinition } from '@console/plugin-sdk';
+import {
+  Plugin,
+  ModelFeatureFlag,
+  ModelDefinition,
+  DashboardsCard,
+  DashboardsTab,
```
Comment (spadgett) [inline frontend/packages/ceph-storage-plugin/src/plugin.ts]: alpha

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resilience/data-resilience.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+      </DashboardCardBody>
+    </DashboardCard>
+  );
+};
+
+export const DataResiliency = withDashboardResources(DataResiliency_);
+
+type DataResiliency = {
+  totalPgRaw: object;
+  cleanAndActivePgRaw: object;
```
Comment (spadgett) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resilience/data-resilience.tsx]: prefer more precise types when possible

**Comment by spadgett** [comment]
/approve

Approving the feature pending code review fixes. Thanks!

**Comment by cloudbehl** [comment]
@rawagner updated the review comments. please have a look.

**Comment by cloudbehl** [comment]
@rawagner Verified the PR

**Pull request merged**